Meadowlark-js/backends/meadowlark-postgresql-backend/src/repository/SqlHelper.ts

```diff
@@ -8,11 +8,16 @@ import format from 'pg-format';
 
 /**
  * Function that returns a SQL query for checking whether references exist for a given list of document ids
+ * This has a secondary function of locking this row for proper updating and deleting
  * @param documentId the document to check for existence
  * @returns SQL query to check for existence ids
  */
 export function existenceIdsForDocument(documentId: string): string {
```

Comment:
I wonder about renaming this so that it is more self-explanatory to the reader. An idea:

```suggestion
export function getReferenceIdentifiers(documentId: string): string {
```

Is that truly better? Not sure. I still don't like the table name `existence`. Now I'm wondering about changing to `references`. But that would be for another ticket, not this one. 

Let's not change the name right now. Leaving this comment in place though, to spur discussion before creating a ticket.

Reply:
I'm not really happy with the name of the existence table either. Something to keep in mind though, we do already have a references table in the Postgres implementation.

Reply:
Ah, I'm getting them confused - because I can never remember what "existence" means! All the more reason to change it. Now I remember that it is for the superclass identification, not for "foreign keys". So changing this function name to use the word "references" would be entirely inappropriate.

Reply:
I created a ticket for it.